modules/apps/redirect/redirect-web/src/main/resources/META-INF/resources/js/RedirectPatterns.js

```diff
@@ -22,14 +22,24 @@ import uuidv4 from 'uuid/v4';
 
 import '../css/redirect_pattern.scss';
 
+const REGEX_URL_ALLOW_RELATIVE = /((([A-Za-z]{3,9}:(?:\/\/)?)|\/(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(https?:\/\/|www.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@.\w_]*)#?(?:[\w]*))((.*):(\d*)\/?(.*))?)/;
+
 const PatternField = ({
-	destinationURL = '',
+	destinationURL: initialDestinationUrl,
 	handleAddClick,
 	handleRemoveClick,
+	handleUrlErrorChange,
 	index,
 	pattern = '',
 	portletNamespace,
 }) => {
+	const [destinationUrl, setDestinationUrl] = useState(initialDestinationUrl);
+	const [urlError, setUrlError] = useState(false);
+
+	const urlAllowRelative = (url) => {
+		return REGEX_URL_ALLOW_RELATIVE && REGEX_URL_ALLOW_RELATIVE.test(url);
+	};
```

Comment:
Can we move this fn outside? We don't need to construct it on each render

Reply:
We can make it shorter too

```js
const urlAllowRelative = (url) => REGEX_URL_ALLOW_RELATIVE?.test(url);
```

Or even, if we don't want to check our own constant.

```js
const urlAllowRelative = (url) => REGEX_URL_ALLOW_RELATIVE.test(url);
```